hypothesis/lms
# Pass page range through to Hypothesis client on assignment launch

When a VitalSource assignment is configured with a page range, pass that through to the Hypothesis client on assignment launch, via the `contentRange` property of the Hypothesis client's configuration.

This change is split into three commits:

1. Improve parsing of VitalSource URLs in `VSBookLocation.from_document_url` so that query params don't end up in the `page` or `cfi` properties of the returned `VSBookLocation` object
2. Move reading of the `vitalsource.page_ranges` setting into `VitalSourceService` so it can be more easily accessed from multiple places
3. Add logic to `JSConfig.add_document_url` to fetch the content range (ie. the page range) for a VS assignment and add it to the Hypothesis client config if the page ranges feature is enabled

**Testing:**

1. Enable the VitalSource page ranges feature for the test application instance
2. Configure a new VitalSource assignment and specify a page range
3. Launch the assignment. The LMS frontend app's configuration should contain a new `hypothesisClient.contentRange = { "start": "start_page", "end": "end_page" }` property
